cmd/backup/script.go

```diff
@@ -159,6 +164,45 @@ func newScript() (*script, error) {
 		}
 	}
 
+	if s.c.SSHHostName != "" {
+		var authMethods []ssh.AuthMethod
```

Comment:
Just for my own understanding, what is the precedence in case you specify multiple methods? Does it go right to left or left to right?

Reply:
I have set up a ssh server with password auth and publickey auth simultaneously. And i debugged app with all cases.

Password | Publickey | Auth
-------- | ---------- | ----------
| :heavy_check_mark: | - | :heavy_check_mark: |
| - | :heavy_check_mark: | :heavy_check_mark: |
| :heavy_check_mark: | ✗ | :heavy_check_mark: |
| ✗ | :heavy_check_mark: | :heavy_check_mark: |
| :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| - | - | ✗ |
| ✗ | - | ✗ |
| - | ✗ | ✗ |
| ✗ | ✗ | ✗ |


When both password and private key wrong error log is something like that:
`panic: newScript: ssh new client: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none password publickey], no supported methods remain`

If we change the password auth method and private key method order in the code. err message changing to:

`panic: newScript: ssh new client: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey password], no supported methods remain`

There is a for loop  that in client_auth.go:64 `for _, a := range config.Auth`

Ssh package firstly tries `none auth method`. Next tries the methods provided order. That means which method provided first it uses according to that order.

Reply:
Sounds perfect, thanks for the thorough explanation.